hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/S3EventsHoodieIncrSource.java

```diff
@@ -189,33 +194,36 @@ public Pair<Option<Dataset<Row>>, String> fetchNextBatch(Option<String> lastCkpt
         .filter(filter)
         .select("s3.bucket.name", "s3.object.key")
         .distinct()
-        .mapPartitions((MapPartitionsFunction<Row, String>)  fileListIterator -> {
+        .rdd()
+        // JavaRDD simplifies coding with collect and suitable mapPartitions signature. check if this can be avoided.
+        .toJavaRDD()
+        .mapPartitions(fileListIterator -> {
```

Comment:
we usually prefer high level dataframe apis. how is it actually beneficial to convert to rdd here? don't quite get the comment